app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DisplayOptionsView.java

```diff
@@ -412,14 +440,22 @@ private boolean setDisplayDensity(float newDensity) {
         return restart;
     }
 
+    private RestartDialogWidget.CancelFunction cancelDisplayDensitySelection(float prevDensity) {
+        RestartDialogWidget.CancelFunction cancelFunction = () -> {
+            setDisplayDensity(prevDensity);
+        };
+
+        return cancelFunction;
+    }
+
     private boolean setDisplayDpi(int newDpi) {
         mBinding.dpiEdit.setOnClickListener(null);
         boolean restart = false;
-        int prevDensity = SettingsStore.getInstance(getContext()).getDisplayDpi();
+        int prevDpi = SettingsStore.getInstance(getContext()).getDisplayDpi();
         if (newDpi < SettingsStore.DISPLAY_DPI_MIN || newDpi > SettingsStore.DISPLAY_DPI_MAX) {
-            newDpi = prevDensity;
+            newDpi = prevDpi;
 
-        } else if (prevDensity != newDpi) {
+        } else if (prevDpi != newDpi) {
```

Comment:
This rename is interesting by itself. It'd be nice if you could split it in a different commit in this PR